lib/TaskProcessing/ChangeToneProvider.php

```diff
@@ -0,0 +1,146 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OCA\OpenAi\TaskProcessing;
+
+use Exception;
+use OCA\OpenAi\AppInfo\Application;
+use OCA\OpenAi\Service\OpenAiAPIService;
+use OCA\OpenAi\Service\OpenAiSettingsService;
+use OCP\IAppConfig;
+use OCP\IL10N;
+use OCP\TaskProcessing\EShapeType;
+use OCP\TaskProcessing\ISynchronousProvider;
+use OCP\TaskProcessing\ShapeDescriptor;
+use OCP\TaskProcessing\ShapeEnumValue;
+use RuntimeException;
+
+class ChangeToneProvider implements ISynchronousProvider {
+
+	public function __construct(
+		private OpenAiAPIService $openAiAPIService,
+		private IAppConfig $appConfig,
+		private OpenAiSettingsService $openAiSettingsService,
+		private IL10N $l,
+		private ?string $userId,
+	) {
+	}
+
+	public function getId(): string {
+		return Application::APP_ID . '-changetone';
+	}
+
+	public function getName(): string {
+		return $this->openAiAPIService->getServiceName();
+	}
+
+	public function getTaskTypeId(): string {
+		return ChangeToneTaskType::ID;
+	}
+
+	public function getExpectedRuntime(): int {
+		return $this->openAiAPIService->getExpTextProcessingTime();
+	}
+
+	public function getInputShapeEnumValues(): array {
+		$toneInputEnumValue = new ShapeEnumValue($this->l->t('Detect language'), 'detect_language');
+		return [
+			'tone_input' => [
+				new ShapeEnumValue($this->l->t('Friendler'), 'friendler'),
+				new ShapeEnumValue($this->l->t('More formal'), 'more formal'),
+				new ShapeEnumValue($this->l->t('Funnier'), 'funnier'),
+				new ShapeEnumValue($this->l->t('More casual'), 'more casual'),
+				new ShapeEnumValue($this->l->t('More urgent'), 'more urgent'),
+			],
+		];
+	}
+
+	public function getInputShapeDefaults(): array {
+		return [
+			'tone_input' => 'friendler',
+		];
+	}
+
+	public function getOptionalInputShape(): array {
+		return [
+			'max_tokens' => new ShapeDescriptor(
+				$this->l->t('Maximum output words'),
+				$this->l->t('The maximum number of words/tokens that can be generated in the completion.'),
```

Comment:
tokens and words are different things. Generally for english, 1 token = 3/4 word. See the link for details: https://gptforwork.com/guides/openai-gpt3-tokens

let's use the term "tokens" exclusively here.

Reply:
I can totally do that, just be aware that this is the same in every single provider right now. Maybe we should stay consistent?

Reply:
ah damn, calling for expert opinion on this @julien-nc

Reply:
The users most likely don't know about the concept of token. Since tokens are approximately words, we could say "approximate number of words". Wdyt?